src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -77,6 +78,58 @@ public String getFqdn() {
         }
     }
 
+    public URI getCasLogoutUri(String doguname) throws GetCasLogoutUriException {
+        JSONObject doguMetaData, properties;
+        try {
+            EtcdKeysResponse.EtcdNode node = getDoguNodeFromEtcd(doguname);
+            doguMetaData = getCurrentDoguNode(node);
+            if (doguMetaData != null) {
+                properties = getPropertiesFromMetaData(doguMetaData);
+            } else {
+                throw new GetCasLogoutUriException("Could not get dogu metadata");
+            }
+            return getLogoutUriFromProperties(properties);
+        } catch (ParseException | URISyntaxException | GetDoguNodeFromEtcdException e) {
+            throw new GetCasLogoutUriException(e.toString());
+        }
+    }
+
+    private URI getLogoutUriFromProperties(JSONObject properties) throws GetCasLogoutUriException, URISyntaxException {
+        Object logoutUri = properties.get("logoutUri");
+        if (logoutUri != null){
+            String logoutUriString = logoutUri.toString();
+            if (logoutUriString != null) {
+                return new URI(logoutUriString);
+            } else {
+                throw new GetCasLogoutUriException("Could not get logoutUri from properties");
+            }
+        } else {
+            throw new GetCasLogoutUriException("Could not get logoutUri from properties");
+        }
+    }
+
+    private JSONObject getPropertiesFromMetaData(JSONObject doguMetaData) throws GetCasLogoutUriException {
+        Object propertiesObject = doguMetaData.get("Properties");
+        if (propertiesObject != null) {
+            if (propertiesObject instanceof JSONObject) {
+                return (JSONObject) propertiesObject;
+            } else {
+                throw new GetCasLogoutUriException("Properties are not in JSONObject format");
+            }
+        } else {
+            throw new GetCasLogoutUriException("No Properties are set");
+        }
+    }
+
+    protected EtcdKeysResponse.EtcdNode getDoguNodeFromEtcd(String name) throws GetDoguNodeFromEtcdException {
+        try {
+            return etcd.getDir("/dogu/"+name).recursive().send().get().getNode();
+        } catch (IOException | EtcdException | EtcdAuthenticationException | TimeoutException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1166)